DOCS-11819-mqtt3-connector-final-kt (#1611)
mulesoft/docs-connectors: mqtt 3/1.0.0/modules/ROOT/pages/mqtt3-connector-quality-of-service.adoc

= Configure the Quality of Service for MQTT 3 Connector

The MQTT protocol enables you to specify levels of delivery assurance for inbound or outbound messages. These levels

> are referred to as quality of service (QoS).

Reviewer: ```suggestion
are referred to as Quality of Service (QoS).
```

The available QoS levels to configure for MQTT 3 Connector are:

 * *AT_MOST_ONCE*
+

> The message will be delivered at most one time (fire and forget). The level does not guarantee that the message will be received at all.

Reviewer: ```suggestion
The message will be delivered at most one time (fire and forget). This level does not guarantee that the message will be received at all.
```

 * *AT_LEAST_ONCE*
+
The message will be delivered correctly at least one time. The message could be delivered more than once.
 * *EXACTLY_ONCE*
+

> The level guarantees that the message will be delivered correctly exactly one time.

Reviewer: ```suggestion
This level guarantees that the message will be delivered correctly exactly one time.
```

> The quality of service can be different for publishing and subscribing, even if the published message and the subscription

Reviewer: ```suggestion
The QoS can be different for publishing and subscribing, even if the published message and the subscription
```
